tvm_executor/src/transaction_executor.rs

```diff
@@ -438,7 +478,7 @@ pub trait TransactionExecutor {
         log::debug!(target: "executor", "acc balance: {:#?}", acc_balance);
         log::debug!(target: "executor", "msg balance: {}", msg_balance.grams);
         let is_ordinary = self.ordinary_transaction();
-        if acc_balance.grams.is_zero() {
+        if acc_balance.grams.is_zero() && !params.is_same_thread_id {
```

Comment:
We just continue if same_thread is set, but how does it affect execution? What will be sent if you want to send all balance, but it must be negative after all phases?